### ISSUE TITLE
Variance and Softmax Loss

### ISSUE THREAD

**Issue by D3lt4lph4** [rykov8/ssd_keras#95]
Hi all,

I am trying to fully understand the code, but there are a few things I am not sure about.

1. What is the purpose of the variance ? I cannot find it in the original paper and don't see why it would be added here. Is it that it has been added in the caffe implementation but not in the paper or did I miss something ? (ssd_layers.py PriorBox or ssd_utils.py decode_boxes for instance)

2. In the ssd_training.py file there is this line in the softmax_loss : `y_pred = tf.maximum(tf.minimum(y_pred, 1 - 1e-15), 1e-15)`. My guess is that the maximum is to avoid the zero value, but what is the point of the minimum ? Is there any problem with having 1 as value ?

**Comment by JordanPeltier** [comment]
Same question for the variance

**Comment by hust-kevin** [comment]
@D3lt4lph4 @JordanPeltier  do you understand the variance? I also have same question

**Comment by Yurodidon** [comment]
The variances is to make the training faster, in the origin paper, authors didn't talk about it, but you can take a look at the authors'  implementation
